src/codegen/replacements.ts

```diff
@@ -54,6 +54,10 @@ export const globalsToPrefix = [
   "undefined",
 ];
 
+// This is a list of symbol we should access using @ notation.
+// Array.prototype.$$iterator -> Array.prototype.__intrinsic_$iterator -> Array.prototype.@$iterator
```

Comment:
At first, I was thinking this should result in `@@iterator`, so that

```
$$iterator -> __intrinsic____intrinsic__ -> @@iterator
```

but I realized you're probably looking to snapshot the original implementations of the Array.prototype.iterator, right?

I kind of wish that the double dollar sign can convert to double at, as [it seems you can refer to any wellknown symbol this way][1], but with the primordials use case this becomes a bit tricky. I guess it comes down to what looks best in our code. I wonder if special casing the primordial protection version just to `{known_class}.prototype.{known_symbol}` might be better. 

[1]: https://github.com/oven-sh/WebKit/blob/0aa1f6dfc9b48be1e89b8f709fd44e3c88feaf33/Source/JavaScriptCore/builtins/ArrayConstructor.js#L55

was also going to suggest looking at the built-in symbols list, like:
```
const symbolPropsToPrefix = Object.getOwnPropertyNames(Symbol)
  .filter(name => typeof Symbol[name] == 'symbol')
  .map(name => `$${name}`)
```

Reply:
> I kind of wish that the double dollar sign can convert to double at, as [it seems you can refer to any wellknown symbol this way](https://github.com/oven-sh/WebKit/blob/0aa1f6dfc9b48be1e89b8f709fd44e3c88feaf33/Source/JavaScriptCore/builtins/ArrayConstructor.js#L55), but with the primordials use case this becomes a bit tricky. I guess it comes down to what looks best in our code. I wonder if special casing the primordial protection version just to {known_class}.prototype.{known_symbol} might be better.

Yes WebKit's `@@iterator` is a reference to the symbol but not the value. So to snapshot the value I've tweaked it a bit. For our magic reference when you to `.$$iterator` we get `.@$iterator` which is the snapshotted value. 
We can write `foo[$$iterator]` to get the live value _(translated into `.@@iterator`).

> was also going to suggest looking at the built-in symbols list, like:

I want to be explicit here since this is partially added by us to WebKit and something we should update as we go whenever new things are added.